src/ext/java/org/opentripplanner/ext/siri/TimetableHelper.java

```diff
@@ -693,6 +685,59 @@ public static TripTimes createUpdatedTripTimes(Timetable timetable, Graph graph,
 
         return newTimes;
     }
+
+    /**
+     * This method maps an ArrivalBoardingActivity to a pick drop type.
+     *
+     * The Siri ArrivalBoardingActivity includes less information than the pick drop type, therefore is it only
+     * changed if routability has changed.
+     *
+     * @param currentValue The current pick drop value on a stopTime
+     * @param arrivalBoardingActivityEnumeration The incoming boardingActivity to be mapped
+     * @return Mapped PickDrop type, empty if routability is not changed.
+     */
+    public static Optional<PickDrop> mapDropOffType(PickDrop currentValue, ArrivalBoardingActivityEnumeration arrivalBoardingActivityEnumeration) {
+        switch (arrivalBoardingActivityEnumeration) {
+            case ALIGHTING:
+                if (currentValue.isNotRoutable()) {
+                    return Optional.of(SCHEDULED);
+                }
+                return Optional.empty();
+            case NO_ALIGHTING:
+                return Optional.of(NONE);
+            case PASS_THRU:
+                return Optional.of(CANCELLED);
+            default:
+                return Optional.empty();
```

Comment:
This is not needed, however we need guard against null before the switch

Reply:
Can this be converted to a Switch expressions? 
``` 
return switch (arrivalBoardingActivityEnumeration) {
    case ALIGHTING -> currentValue.isNotRoutable() ? Optional.of(SCHEDULED) : Optional.empty();
    case NO_ALIGHTING -> Optional.of(NONE);
    :
```

Reply:
I have added a null check and a corresponding unit test.

I have also changed this to use switch expressions.